cache/memcached_client.go

```diff
@@ -587,47 +512,23 @@ func (c *memcachedClient) trackError(op string, err error) {
 	var netErr net.Error
 	switch {
 	case errors.As(err, &connErr):
-		c.failures.WithLabelValues(op, reasonTimeout).Inc()
+		c.metrics.failures.WithLabelValues(op, reasonTimeout).Inc()
 	case errors.As(err, &netErr):
 		if netErr.Timeout() {
-			c.failures.WithLabelValues(op, reasonTimeout).Inc()
+			c.metrics.failures.WithLabelValues(op, reasonTimeout).Inc()
 		} else {
-			c.failures.WithLabelValues(op, reasonNetworkError).Inc()
+			c.metrics.failures.WithLabelValues(op, reasonNetworkError).Inc()
 		}
 	case errors.Is(err, memcache.ErrMalformedKey):
-		c.failures.WithLabelValues(op, reasonMalformedKey).Inc()
+		c.metrics.failures.WithLabelValues(op, reasonMalformedKey).Inc()
 	case errors.Is(err, memcache.ErrServerError):
-		c.failures.WithLabelValues(op, reasonServerError).Inc()
-	default:
-		c.failures.WithLabelValues(op, reasonOther).Inc()
-	}
-}
-
-func (c *memcachedClient) enqueueAsync(op func()) error {
-	select {
-	case c.asyncQueue <- op:
-		return nil
+		c.metrics.failures.WithLabelValues(op, reasonServerError).Inc()
 	default:
-		return errMemcachedAsyncBufferFull
-	}
-}
-
-func (c *memcachedClient) asyncQueueProcessLoop() {
-	defer c.workers.Done()
-
-	for {
-		select {
-		case op := <-c.asyncQueue:
-			op()
-		case <-c.stop:
-			return
-		}
+		c.metrics.failures.WithLabelValues(op, reasonOther).Inc()
 	}
 }
 
 func (c *memcachedClient) resolveAddrsLoop() {
-	defer c.workers.Done()
```

Comment:
We lost this logic. We're not waiting for `resolveAddrsLoop()` go routine to end when we call `memcachedClient.Stop()`. However, I think it's not a problem.